src/_pytest/logging.py

```diff
@@ -575,6 +582,18 @@ def __init__(self, config: Config) -> None:
             get_option_ini(config, "log_auto_indent"),
         )
         self.log_cli_handler.setFormatter(log_cli_formatter)
+        if config.option.suppress_logger:
+            self._suppress_loggers(config)
+
+    @staticmethod
+    def _suppress_loggers(config: Config) -> None:
+        logger_names = set(config.option.suppress_logger)
+        for name in logger_names:
+            # disable propagation for the given logger preventing of event passing to ancestor handlers
+            # adding a NullHandler to prevent logging.LastResort handler from logging warnings to stderr
+            logger = logging.getLogger(name)
+            logger.addHandler(logging.NullHandler())
```

Comment:
unsure about this

Reply:
Not sure myself, but what happens to other handlers that are installed in the logger? I suppose they will still process messages?

Reply:
Interesting one, from a quick check it will still use such handlers, this is outlined below:

```
>>> import logging
>>> log = logging.getLogger('pytest')
>>> log.setLevel(logging.DEBUG)
>>> # add a stream handler to stderr by default
>>> log.addHandler(logging.StreamHandler())
>>> log.debug('stderr')
stderr
>>> log.addHandler(logging.NullHandler())
>>> log.debug('does this still write to stderr')
does this still write to stderr
```

two questions:

- Should we iterate / remove the handlers on the given logger?
- Should we check the hierarchy of loggers if the logger provides exists, else I assume its going to 'create' one by badly typed --suppress-logger names etc (maybe thats not a big deal)

Reply:
```
#1
check name is in: 
logs = [logging.getLogger(name).name for name in logging.root.manager.loggerDict] ?

#2
log.handlers = [] upfront perhaps?

Reply:
What we do to suppress logs of certain loggers is to simply set the logLevel of those loggers to `CRITICAL`. Have you thought about doing that?

Reply:
No I haven’t really, but it could be viable tho still showing certain levels, would this setting perhaps be better named and behaving as a means to set log levels on individual loggers, rather than attempting to fully suppress them?

Reply:
The more I think about it, it does make sense, maybe even for ERROR too?